modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1916,6 +1908,15 @@ protected void invalidated() {
                     pseudoClassStateChanged(DISABLED_PSEUDOCLASS_STATE, get());
                     updateCanReceiveFocus();
                     focusSetDirty(getScene());
+
+                    if (Node.this instanceof Parent) {
+                        List<Node> children = ((Parent)Node.this).getChildren();
+                        if (children != null) {
+                            for (Node child : children) {
+                                child.updateDisabled();
+                            }
+                        }
+                    }
```

Comment:
Since `getChildren()` is not final, one can easily override it and return null. 
Therefore, this check should still be done here. 
Maybe we even need to check that every child is not null, since again I can override `getChildren()` and return a list that allows null. (The default implementation in `Parent` throws an exception when adding a null child)

Reply:
It's definitely a hypothetical case. `getChildren()` is called all over the place in JavaFX without a null check, so I see no reason for null checks here.

Reply:
Technically correct, although the [docs of `getChildren()`](https://openjfx.io/javadoc/18/javafx.graphics/javafx/scene/Parent.html#getChildren()) specifically says:

> Note to subclasses: if you override this method, you must return from your implementation the result of calling this super method. The actual list instance returned from any getChildren() implementation must be the list owned and managed by this Parent. The only typical purpose for overriding this method is to promote the method to be public.

So considering the case that an overriding method will return `null` is not practical.

As for a `null` child, I think that that's also not allowed. Considering that a child can be added at most once to a scenegraph, it would mean that only 1 `null` child is allowed. There might be more issues with `null` children. I don't think there's a good reason to check for them.

Reply:
Yep, right. While it is possible to override `getChildren()` and return null, it will break everywhere so we can also omit the null check here.